lib/pippi/checks/method_sequence_checker.rb

```diff
@@ -18,14 +19,21 @@ def decorate
       second_method_decorator = if method_sequence_check_instance.check_descriptor.method_sequence.decorator
         method_sequence_check_instance.check_descriptor.method_sequence.decorator
       else
+        # Create a new module which will be used to intercept method calls via Object#extend
         Module.new do
           descriptor = method_sequence_check_instance.check_descriptor.method_sequence.method2
+          # Define a new method which will be used to intercept & decorate the original method call
+          # Delegate to super after calling decorator
           define_method(descriptor) do |*args, &blk|
             # Using "self.class" implies that the first method invocation returns the same type as the receiver
             # e.g., Array#select returns an Array.  Would need to further parameterize this to get
             # different behavior.
+            #
+            # TODO jbodah - We could use a closure here instead of an instance
+            #   variable which might solve the issue noted above
             the_check = self.class.instance_variable_get(name)
             the_check.add_problem
+            # TODO - @tcopeland could you explain this block?
```

Comment:
Hey sorry for the delay in replying @jbodah.

The problem this is trying to solve is illustrated in this sequence of method calls:

```
x = foo.select {|y| y > 2 }
z = x.first
x.each {|q| puts q }
```

Normally pippi would flag that code since `select` is called and then `first` is called on the result.  But in this case, the result is also used as a collection later - that is, we call `each` on it.   So the code can't be simplified down into a single call to `detect`.  

Pippi's logic is more or less to add a `Problem`, and then remove it if needed.  There might be a nicer way to do that... like, adding it as a provisional problem and not actually adding it to the report until we confirm it.